### DISCUSSION TITLE
Add storage option to use-local-storage hook

### DISCUSSION THREAD

**Discussion by GabriFila** [mantinedev/mantine#2273] [Feature requests]
Hi everyone, loving mantine hooks!

I used the [useLocalStorage](https://mantine.dev/hooks/use-local-storage/#browser-tabs-synchronization) hook, I like ita lot but I would love to use the same functionality with the `SessionStorage` not only `LocalStorage`.

In that sense I propose to add a `storage` prop that can be `'session'` or `'local'` to allow the user to specify the target storage, it could default to one of the two if not specified. The hook would internally be responsible to also move/cleanup the data from one storage to the other when the `session` prop changes.

What do you think about it?

I'd be interested in implementing this feature!

Thank you

**Comment by zhulien-ivanov** [comment]
You can follow #2214 for the progress on `useSessionStorage` as it is already implemented and awaits potential merge.

**Reply by GabriFila** [reply]
Thank you!
